06-descriptive-analysis-in-srvyr.Rmd

```diff
@@ -1 +1,420 @@
+  
+We should use different measures depending on whether our variables are interval/ratio, ordinal, or nominal.
+
+## Descriptive analysis using the {srvyr} package
+
+### Setup
+
+Recall from [Chapter 05](#c05) the general process for estimation with the {srvyr} package:
+
+1. Create a `tbl_svy` object using `srvyr::as_survey_design()` or `srvyr::as_survey_rep()`.
+2. Subset the data for subpopulations using `dplyr::filter()`, if needed.
+3. Specify domains of analysis using `dplyr::group_by()`, if needed.
+4. Within `srvyr::summarize()`, specify variables to calculate ,means, totals, proportions, quantiles, and more.
+
+With the ANES data, we create a `tbl_svy` object using `srvyr::as_survey_design()`,  adjusting the weight to add up to the citizen population:
+
+```{r anes_des}
+#| error: FALSE
+#| warning: FALSE
+#| message: FALSE
+library(survey) # for survey analysis
+library(srvyr) # for tidy survey analysis
+library(readr)
+library(here)
+
+anes <-
+  read_rds(here::here(
+    "data", "anes_2020.rds"
+  )) %>%
+  mutate(Weight = Weight / sum(Weight) * 231592693)
+
+anes_des <- anes %>%
+  as_survey_design(
+    weights = Weight,
+    strata = Stratum,
+    ids = VarUnit,
+    nest = TRUE
+  )
+```
+
+### Count observations using survey methods with `survey_count()`
+
+With `srvyr::survey_count()`, you can produce weighted counts and variance of your choice. The syntax is very similar to the `dplyr::count()` syntax. However, it can only be called on `tbl_srvy()` objects. Let's explore the syntax:
+
+```r
+survey_count(
+  x,
+  ...,
+  wt = NULL,
+  sort = FALSE,
+  name = "n",
+  .drop = dplyr::group_by_drop_default(x),
+  vartype = c("se", "ci", "var", "cv")
+  )
+```
+
+The arguments are:
+
+* `x`: a `tbl_svy` object created by `as_survey`
+* `...`: variables to group by, passed to `group_by`
+* `wt`: a variable to weight on in addition to the survey weights, defaults to `NULL`
+* `sort`: how to sort the variables, defaults to `FALSE`
+* `name`: the name of the count variable, defaults to `n`
+* `.drop`: whether to drop empty groups
+* `vartype`: type(s) of variation estimate to calculate, defaults to `se` (standard error)
+
+The steps to use `survey_count()` are:
+
+* Specify the sample design,
+* Filter subsets using `dplyr::filter()`, if needed
+* Run `survey_count()`, specifying the required arguments within the function
+
+Let's see the weighted count of responses in ANES:
+
+```{r}
+anes_des %>% #  Specify the sample design
+  survey_count() # Run `survey_count()`
+```
+
+`srvyr::count()` can take one or many variables. To calculate a cross-tab of population in each age group and gender, we run the below:
+
+```{r}
+anes_des %>%
+  # Specify the required arguments within the function
+  survey_count(AgeGroup, Gender, name = "N") 
+```
+
+### Calculate totals using survey methods using `survey_total()`
```

Comment:
Need an example of how this is different from `survey_count`. No good examples from ANES. For RECS, maybe something like the total amount of money spent on electricity by region?